src/NBitcoin/NetworkOptions.cs

```diff
@@ -31,11 +45,53 @@ public NetworkOptions(uint flags = All)
             this.flags = flags;
         }
 
+        /// <summary>
+        /// Get/Set ProofOfStake flags.
+        /// </summary>
+        public bool IsProofOfStake
+        {
+            get
+            {
+                bool isPOS = (this.flags & POS) != 0;
+                // This sanity check will be removed once the static flags are removed
+                if (isPOS != Block.BlockSignature)
+                {
+                    throw new ArgumentException($"Block.BlockSignature { Block.BlockSignature} mismatches cross-check value: { isPOS }");
+                }
+                // This sanity check will be removed once the static flags are removed
+                if (isPOS != Transaction.TimeStamp)
+                {
+                    throw new ArgumentException($"Transaction.TimeStamp {Transaction.TimeStamp} mismatches cross-check value: { isPOS }");
+                }
+                return isPOS;
+            }
+
+            set
+            {
+                this.flags = value ? (this.flags | POS) : (this.flags & ~POS);
+            }
+        }
+
+        /// <summary>
+        /// Get/Set Witness flag.
+        /// </summary>
+        public bool IsWitness
+        {
+            get
+            {
+                return (this.flags & Witness) != 0;
+            }
+
+            set
+            {
+                this.flags = value ? (this.flags | Witness) : (this.flags & ~Witness);
+            }
+        }
         /// <summary>
         /// Clones the NetworkOptions object.
         /// </summary>
         /// <returns>The cloned NetworkOptions object.</returns>
-        public NetworkOptions Clone()
+        public virtual NetworkOptions Clone()
```

Comment:
If you want this to allow inheritance you will need to create a `protected virtual OnClone()` method
Not this PR though

Reply:
Ok